atlasdb-api/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -230,6 +231,9 @@ public void deregisterTaggedMetrics(Predicate<MetricName> predicate) {
                     .collect(Collectors.toList());
 
             metricsToRemove.forEach(taggedMetricRegistry::remove);
+
+            // Don't be clever, invalidate everything. This should be relatively uncommon.
+            metricNameCache.invalidate();
```

Comment:
seems fine, but I am curious, do we really ever need to invalidate this cache?

Reply:
Need to? Probably not, but there's a risk of a memory leak if we remove data from the TaggedMetricRegistry without removing it from the cache.

Based on callers to this method, we could remove cache invalidation, but I don't think it buys us enough to be worth sacrificing safety.

Reply:
Yeah, that's fair.